packages/amplify_flutter/lib/amplify_hub.dart

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:amplify_core/types/hub/HubEvent.dart';
+import 'amplify.dart';
+
+typedef void Listener(dynamic event);
+
+class AmplifyHub {
+
+  final Map<HubChannel, Stream> availableStreams = {};
+
+  /// Expose listen method which instantiates new StreamController listening to one or more availableStreams
+  StreamSubscription listen(List<HubChannel> channels, @required Listener listener) {
+
+    List<StreamSubscription> platformSubscriptions = [];
+    StreamController controller;
+
+    listenToPluginStreams() {
+      channels.forEach((c) {
+        if (availableStreams[c] != null) {
+          StreamSubscription subscription = availableStreams[c].listen((msg) {
+            /// Emit events via Hub
+            controller.add(msg);
+          });
+          platformSubscriptions.add(subscription);
+        } else {
+          print('You are attempting to listen to a plugin that has not been added.');
+        }
+      });
+    }
```

Comment:
didn't you already put this inline in `onListen()` for `StreamController.broadcast`?

Reply:
@Amplifiyer What exactly do you mean by "this"?  The code above is listening to the streams that are established in StreamController.broadcast, and re-broadcasting them on the Hub Stream to which the customer is listening.

Reply:
but who is calling this `listenToPluginStreams`. Reading the code it seems it was supposed to be used for `onListen` on line 55 of this file as in `onListen: listenToPluginStreams` but it seems you are again using the same code inline from line 56 to 66 of this file.

Reply:
ah geez, I see what you're saying.  I think this was a merge error. Yes, it's duplicative code.